src/app/query/stacks/mempool/mempool.hooks.ts

```diff
@@ -36,13 +36,46 @@ function useAccountUnanchoredMempoolTransactions(address: string) {
   }, [txs]);
 }
 
+function useAccountUnanchoredMempoolTransactionsQuery(address: string) {
+  const query = useAccountMempool(address);
+  const accountMempoolTxs = query.data;
+  const mempoolTxs = (accountMempoolTxs ? accountMempoolTxs.results : []) as MempoolTransaction[];
+  const results = mempoolTxs.filter(
+    tx => tx.tx_status === 'pending' && !droppedCache.has(tx.tx_id)
+  );
+  const txs = useTransactionsById(results.map(tx => tx.tx_id));
+  return useMemo(() => {
+    const newData = txs
+      .map(tx => tx.data)
+      .filter(tx => {
+        if (typeof tx === 'undefined') return false;
+        if (droppedCache.has(tx.tx_id)) return false;
+        if (tx.tx_status !== 'pending') {
+          // Stale txs persist in the mempool endpoint so we
+          // need to cache dropped txids to prevent unneeded fetches
+          droppedCache.set(tx.tx_id, true);
+          return false;
+        }
+        return true;
+      });
+    return { ...query, data: newData };
+  }, [txs, query]);
+}
+
 export function useStacksPendingTransactions() {
   const address = useCurrentAccountStxAddressState();
   return useAccountUnanchoredMempoolTransactions(address ?? '').filter(
     tx => !!tx
   ) as MempoolTransaction[];
 }
 
+export function useStacksPendingTransactionsQuery() {
+  const address = useCurrentAccountStxAddressState();
+  const query = useAccountUnanchoredMempoolTransactionsQuery(address ?? '');
+  const newData = query.data.filter(tx => !!tx) as MempoolTransaction[];
+  return { ...query, data: newData };
```

Comment:
Returns a new referential equality on every render

Reply:
To keep the same reference to `query` we can go for `query.data = newData` together with some type gymnastics to set the appropriate type in order to conserve the remaining query properties present in the `query` object. Alternatively, we can pick out a few properties from the query object, say `isInitialLoading`, `data`, and `error` to name a few, although in that case we'll always be coming back to this hook to edit it's return value if a consumer ever needs more data from within the `query` object (which may be fine). So in this latter scenario the return value would change depending on what is needed by consumers.

Alternatively all of this can be moved into the component itself to have direct access to `query` and then we don't have to deal with the complexities of this intermediate hook, at the expense of making the component larger. 

What do you reckon is best? Maybe there are other options not described here?

Reply:
Would avoid mutating the object.

I'd investigate either:
- parsing the response by passing in a configurable `select` to the query to handle the parsing
- return the data as a `useMemo`

Reply:
After investigating a little I came across `useAccountUnanchoredMempoolTransactions` that complicates refactoring due to its side effects and use of external state: the state and query object live in different files, so getting query data to the component that needs it would require significant refactoring.

@kyranjamie and i settled for not refactoring this too much as the implications of these side effects are not trivial to follow, and opting for analytics to keep track of how often the side effect occurs.